Sources/NIOHTTP1/HTTPEncoder.swift

```diff
@@ -190,3 +182,377 @@ public final class HTTPResponseEncoder: ChannelOutboundHandler {
         }
     }
 }
+
+private extension ByteBuffer {
+    private mutating func write(status: HTTPResponseStatus) {
+        if case .ok = status {
+            // Optimize for 200 ok, which should be the most likely code (...hopefully).
+            self.write(staticString: status200)
+        } else {
+            self.write(string: String(status.code))
+            self.writeWhitespace()
+            self.write(string: status.reasonPhrase)
+        }
+    }
+
+    mutating func write(response: HTTPResponseHead) {
+        // Optimize for HTTP/1.0 and HTTP/1.1
+        if response.version.major == 1 {
+            if response.version.minor == 1 {
+                switch response.status {
```

Comment:
Is this better written as a single switch statement, rather than two? E.g. `switch (response.version.major, response.version.minor, status)`?

Reply:
Yes! TIL :)